src/bidiMapper/domains/storage/StorageProcessor.ts

```diff
@@ -0,0 +1,283 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type {Protocol} from 'devtools-protocol';
+
+import type {CdpClient} from '../../../cdp/CdpClient.js';
+import type {Storage} from '../../../protocol/protocol.js';
+import {
+  InvalidArgumentException,
+  Network,
+  UnderspecifiedStoragePartitionException,
+  UnableToSetCookieException,
+  UnsupportedOperationException,
+} from '../../../protocol/protocol.js';
+import {assert} from '../../../utils/assert.js';
+import type {LoggerFn} from '../../../utils/log.js';
+import {LogType} from '../../../utils/log.js';
+import type {BrowsingContextStorage} from '../context/BrowsingContextStorage.js';
+import {NetworkProcessor} from '../network/NetworkProcessor.js';
+
+/**
+ * Responsible for handling the `storage` domain.
+ */
+export class StorageProcessor {
+  readonly #browserCdpClient: CdpClient;
+  readonly #browsingContextStorage: BrowsingContextStorage;
+  readonly #logger: LoggerFn | undefined;
+
+  constructor(
+    browserCdpClient: CdpClient,
+    browsingContextStorage: BrowsingContextStorage,
+    logger: LoggerFn | undefined
+  ) {
+    this.#browsingContextStorage = browsingContextStorage;
+    this.#browserCdpClient = browserCdpClient;
+    this.#logger = logger;
+  }
+
+  async getCookies(
+    params: Storage.GetCookiesParameters
+  ): Promise<Storage.GetCookiesResult> {
+    const partitionKey = this.#expandStoragePartitionSpec(params.partition);
+
+    const cdpResponse = await this.#browserCdpClient.sendCommand(
+      'Storage.getCookies',
+      {}
+    );
+
+    const filteredBiDiCookies = cdpResponse.cookies
+      .filter(
+        // CDP's partition key is the source origin.
+        (c) =>
+          c.partitionKey === undefined ||
+          c.partitionKey === partitionKey.sourceOrigin
+      )
+      .map((c) => this.#cdpToBiDiCookie(c))
+      .filter((c) => this.#matchCookie(c, params.filter));
+
+    return {
+      cookies: filteredBiDiCookies,
+      partitionKey,
+    };
+  }
+
+  async setCookie(
+    params: Storage.SetCookieParameters
+  ): Promise<Storage.SetCookieResult> {
+    const partitionKey = this.#expandStoragePartitionSpec(params.partition);
+
+    if (params.cookie.value.type !== 'string') {
+      // CDP supports only string values in cookies.
+      throw new UnsupportedOperationException(
+        'Only string cookie values are supported'
+      );
+    }
+    const deserializedValue = params.cookie.value.value;
+
+    try {
+      await this.#browserCdpClient.sendCommand('Storage.setCookies', {
+        cookies: [
+          {
```

Comment:
Maybe create a helper to convert bidiCookiesToCdpCookies?

Reply:
Good idea! Done